src/lib/prep.ml

```diff
@@ -177,25 +179,43 @@ module Prep = struct
     Capnp_rpc_lwt.Capability.with_ref build_job @@ fun build_job ->
     let** _ = Current_ocluster.Connection.run_job ~job build_job in
     (* extract result from logs *)
-    let extract_hashes (git_hashes, failed) line =
+    let extract_hashes (git_hashes, failed, retriable_errors) line =
       match Storage.parse_hash ~prefix:"HASHES" line with
-      | Some value -> (value :: git_hashes, failed)
+      | Some value -> (value :: git_hashes, failed, retriable_errors)
       | None -> (
           match String.split_on_char ':' line with
           | [ prev; branch ] when Astring.String.is_suffix ~affix:"FAILED" prev ->
               Current.Job.log job "Failed: %s" branch;
-              (git_hashes, branch :: failed)
-          | _ -> (git_hashes, failed))
+              (git_hashes, branch :: failed, retriable_errors)
+          | [ prev; branch ] when Astring.String.is_infix ~affix:"Temporary failure" prev ->
+              (git_hashes, failed, branch :: retriable_errors)
+          | _ -> (git_hashes, failed, retriable_errors))
     in
 
-    let** git_hashes, failed = Misc.fold_logs build_job extract_hashes ([], []) in
-    Lwt.return_ok
-      ( List.map
-          (fun (r : Storage.id_hash) ->
-            Current.Job.log job "%s -> %s" r.id r.hash;
-            r)
-          git_hashes,
-        failed )
+    let** git_hashes, failed, retriable_errors = Misc.fold_logs build_job extract_hashes ([], [], []) in
+    if retriable_errors = []
+      then
+        Lwt.return_ok
+          ( List.map
+              (fun (r : Storage.id_hash) ->
+                Current.Job.log job "%s -> %s" r.id r.hash;
+                r)
+              git_hashes,
+            failed )
+      else
+
+        Lwt_unix.sleep 30.0 >>= fun () ->
```

Comment:
A random sleep between 20 and 40 seconds might be useful to spread out concurrent connections.